src/Common/src/Interop/Unix/System.Native/Interop.ReadLink.cs

```diff
@@ -16,9 +18,47 @@ internal static partial class Sys
         /// <param name="buffer">The buffer to hold the output path</param>
         /// <param name="bufferSize">The size of the buffer</param>
         /// <returns>
-        /// Returns the number of bytes placed into the buffer on success; otherwise, -1 is returned
+        /// Returns the number of bytes placed into the buffer on success; 0 if the buffer is too small; and -1 on error.
         /// </returns>
         [DllImport(Libraries.SystemNative, EntryPoint = "SystemNative_ReadLink", SetLastError = true)]
-        internal static extern unsafe int ReadLink(string path, byte[] buffer, int bufferSize);
+        private static extern unsafe int ReadLink(string path, byte[] buffer, int bufferSize);
+
+        /// <summary>
+        /// Takes a path to a symbolic link and returns the link target path.
+        /// </summary>
+        /// <param name="path">The path to the symlink</param>
+        /// <returns>
+        /// Returns the link to the target path on success; and null otherwise.
+        /// </returns>
+        public static string ReadLink(string path)
+        {
+            int bufferSize = 256;
+            do
+            {
+                byte[] buffer = ArrayPool<byte>.Shared.Rent(bufferSize);
+                try
+                {
+                    int resultLength = Interop.Sys.ReadLink(path, buffer, buffer.Length - 1); // -1 for null termination
+                    if (resultLength > 0)
+                    {
+                        // success, null terminate
+                        buffer[resultLength] = (byte)'\0';
```

Comment:
I realize this was here before (and I was likely the one to put it there), but is it actually necessary?  Encoding.UTF8.GetString is told exactly how many bytes to look at, and this one isn't included.

Reply:
(in which case the - 1 in buffer.Length - 1 shouldn't be necessary, either)

Reply:
Yes, we can remove that.